pos_pricelist/static/src/js/models.js

```diff
@@ -73,60 +73,6 @@ function pos_pricelist_models(instance, module) {
         }
     });
 
-    /**
-     * Extend the order
-     */
-    module.Order = module.Order.extend({
-        /**
-         * override this method to merge lines
-         * TODO : Need some refactoring in the standard POS to Do it better
-         * TODO : from line 73 till 85, we need to move this to another method
-         * @param product
-         * @param options
-         */
-        addProduct: function (product, options) {
-            options = options || {};
-            var attr = JSON.parse(JSON.stringify(product));
-            attr.pos = this.pos;
-            attr.order = this;
-            var line = new module.Orderline({}, {
-                pos: this.pos,
-                order: this,
-                product: product
-            });
-            var self = this;
-            var found = false;
-
-            if (options.quantity !== undefined) {
-                line.set_quantity(options.quantity);
-            }
-            if (options.price !== undefined) {
-                line.set_unit_price(options.price);
-            }
-            if (options.discount !== undefined) {
-                line.set_discount(options.discount);
-            }
-
-            var orderlines = [];
-            if (self.get('orderLines').models !== undefined) {
-                orderlines = self.get('orderLines').models;
-            }
-            for (var i = 0; i < orderlines.length; i++) {
```

Comment:
I think it's this hidden feature: upstream Odoo only considers the last line to merge with, this change considers all lines on the order. It should probably go into OCB instead.

Reply:
yep, it's not the place. (and overwrite function is allways a good way to generate bugs).

Ideally, it should be an option. In some cases, users want to merge all lines, sometime no lines. (but it's another story).

thanks for the review @StefanRijnhart.